packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin.swift

```diff
@@ -1,19 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
 import Flutter
-import UIKit
-
-public class AmplifyPushNotificationsPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "amplify_push_notifications", binaryMessenger: registrar.messenger())
-    let instance = AmplifyPushNotificationsPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
-
-  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    switch call.method {
-    case "getPlatformVersion":
-      result("iOS " + UIDevice.current.systemVersion)
-    default:
-      result(FlutterMethodNotImplemented)
+import amplify_flutter_ios
+import AmplifyUtilsNotifications
+
+public class AmplifyPushNotificationsPlugin: NSObject, FlutterPlugin, PushNotificationsHostApi {
+    var remoteNotificationCompletionHandlers: [String: (UIBackgroundFetchResult) -> Void] = [:]
+    // Set to true if the App was awaken in the background by a remote notification from the terminated state
+    var awokeInBackgroundButHasNotLaunched = false
+
+    let sharedEventsStreamHandlers: EventsStreamHandlers
+    let flutterApi: PushNotificationsFlutterApi
+
+    private init(
+        sharedEventsStreamHandlers: EventsStreamHandlers,
+        flutterApi: PushNotificationsFlutterApi
+    ) {
+        self.sharedEventsStreamHandlers = sharedEventsStreamHandlers
+        self.flutterApi = flutterApi
+    }
+
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let messenger = registrar.messenger()
+        let pluginInstance = AmplifyPushNotificationsPlugin(
+            sharedEventsStreamHandlers: EventsStreamHandlers(binaryMessenger: messenger),
+            flutterApi: PushNotificationsFlutterApi(binaryMessenger: messenger)
+        )
+
+        PushNotificationsHostApiSetup(messenger, pluginInstance)
+
+        registrar.addApplicationDelegate(pluginInstance)
+    }
+
+    public func getPermissionStatus(completion: @escaping (GetPermissionStatusResult?, FlutterError?) -> Void) {
+        Task {
+            switch await AUNotificationPermissions.status {
```

Comment:
If I call `getPermissionStatus` before calling `requestPermissionStatus` it should return `.notRequested`, no?